align/schema/constraint.py

```diff
@@ -420,19 +418,49 @@ def max_height_validator(cls, value):
         assert value >= 0, f'Boundary:max_height should be greater than zero {value}'
         return value
 
-    def check(self, checker):
-        bvar = checker.bbox_vars(self.subcircuit, is_subcircuit=True)
+    def translate(self, solver):
+        bvar = solver.bbox_vars('subcircuit')
         if self.max_width is not None:
-            yield checker.cast(bvar.urx-bvar.llx, float) <= 1000*self.max_width  # in nanometer
+            yield solver.cast(bvar.urx-bvar.llx, float) <= 1000*self.max_width  # in nanometer
         if self.max_height is not None:
-            yield checker.cast(bvar.ury-bvar.lly, float) <= 1000*self.max_height  # in nanometer
+            yield solver.cast(bvar.ury-bvar.lly, float) <= 1000*self.max_height  # in nanometer
 
+class GroupBlocks(HardConstraint):
```

Comment:
This is an okay start. We might not want a group to always to form a rectangular bounding box around the containing blocks but we can fix that when we have an example.

Reply:
Agreed